render.lisp

```diff
@@ -320,18 +320,31 @@
             (cl-cairo2:fill-path)
             ))))
 
+(defun init-image-cursor ()
+  (unless *default-cursor*
+    (setf *default-cursor* (make-instance 'cairo-surface
+                                          :allow-gl t
+                                          :filename "assets/cursor.png"))))
+
+(defun make-g-pt-quad (top bottom left right)
+  `((,(cepl:v! left top 0)     ,(cepl:v! 0 0))
+    (,(cepl:v! left bottom 0)  ,(cepl:v! 0 1))
+    (,(cepl:v! right top 0)    ,(cepl:v! 1 0))
+    (,(cepl:v! right bottom 0) ,(cepl:v! 1 1))
+    (,(cepl:v! right top 0)    ,(cepl:v! 1 0))
+    (,(cepl:v! left bottom 0)  ,(cepl:v! 0 1))))
+
 (defmethod draw-cursor ((cursor (eql nil)) fbo x y ortho)
   (unless *default-cursor*
-    (init-cursor)
+    (init-image-cursor)
```

Comment:
TBH I'd recommend sticking to INIT-VECTOR-CURSOR here. We don't have a proper pixel-art of the pointer and I made it look for the image file using relative paths which is only good for the proof-of-implementation.

Still looks better than a plain white triangle. :)